src/VisualStudio/Core/Def/SampleToolWindow/SampleToolboxUserControl.xaml.cs

```diff
@@ -0,0 +1,460 @@
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.QuickInfo;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Extensions;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Extensions;
+using Microsoft.VisualStudio.LanguageServices.Setup;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.VisualStudio.LanguageServices
+{
+    using Workspace = Microsoft.CodeAnalysis.Workspace;
+
+    /// <summary>
+    /// Interaction logic for SampleToolboxUserControl.xaml
+    /// </summary>
+    internal partial class SampleToolboxUserControl : UserControl, IOleCommandTarget
+    {
+        public SampleToolboxUserControl(Workspace workspace, IDocumentTrackingService documentTrackingService, ILanguageServiceBroker2 languageServiceBroker, IThreadingContext threadingContext)
+        {
+            InitializeComponent();
+            InitializeIfNeeded(workspace, documentTrackingService, languageServiceBroker, threadingContext);
+        }
+
+        private Workspace? workspace { get; set; }
+        private DocumentId? lastDocumentId { get; set; }
+        private List<DocSymbol>? originalTree { get; set; }
+        private ITextSnapshot? snapshot { get; set; }
+        private IWpfTextView? textView { get; set; }
+
+        private void InitializeIfNeeded(Workspace workspace, IDocumentTrackingService documentTrackingService, ILanguageServiceBroker2 languageServiceBroker, IThreadingContext threadingContext)
+        {
+            this.workspace = workspace;
+            documentTrackingService.ActiveDocumentChanged += DocumentTrackingService_ActiveDocumentChanged;
+
+            this.textView = GetActiveTextView();
+            if (this.textView is not null)
+            {
+                this.textView.Caret.PositionChanged += FollowCursor;
+            }
+
+            searchBox.GotFocus += RemoveText;
+            searchBox.LostFocus += AddText;
+
+            async void DocumentTrackingService_ActiveDocumentChanged(object sender, DocumentId? documentId)
```

Comment:
I believe `async void` is what the IDE is going to suggest to you here, unfortunately this is not the right thing to do in Visual Studio.

I would change it to just be `void DocumentTrackingService_ActiveDocumentChanged(object sender, DocumentId? documentId)`


and inside wrap your entire method body in:

```csharp
threadingContext.JoinableTaskFactory.RunAsync(async () =>
{
	// Existing method code goes here.
}).FileAndForget("Document Outline: Active Document Changed");
```